content/storage/docs/spark/instance-store.md

```diff
@@ -0,0 +1,146 @@
+
+However, please note that if you have multiple NVMe disks attached to the instances, it will be required to create a RAID 0 setup of all the disks before mounting the `/var/lib/kubelet/pods` directory on the RAID partition. Without a RAID setup, it will not be possible to leverage all the disks capacity available on the node.
+
+The following example shows how to create a node group in your cluster that leverages this approach. In order to prepare our NVMe disks, we can use the [eksctl](https://eksctl.io/) **preBootstrapCommands** definition while creating the node group to execute additional commands during the boot process of the instance. The script will perform the following actions:
+
+* For instances with a single NVMe disk, format the filesystem, create a Linux partition (e.g. ext4, xfs, etc.)
+* For instances with multiple NVMe disks, create a RAID 0 setup across all available disks
+
+Once the disks have been formatted and ready to use, we can then mount the folder **/var/lib/kubelet/pods** using the filesystem just created and set the correct permissions. Below you can find an example of an eksctl configuration file to create a managed node group using this approach.
+
+**Example**
+
+```
+apiVersion: eksctl.io/v1alpha5
+kind: ClusterConfig
+
+metadata:
+  name: YOUR_CLUSTER_NAME
+  region: YOUR_REGION
+
+managedNodeGroups:
+  - name: ng-c5d-9xlarge
+    instanceType: c5d.9xlarge
+    desiredCapacity: 1
+    privateNetworking: true
+    subnets:
+      - YOUR_NG_SUBNET
+    preBootstrapCommands: # commands executed as root
+      - yum install -y mdadm nvme-cli
+      - nvme_disks=($(nvme list | grep "Amazon EC2 NVMe Instance Storage" | awk -F'[[:space:]][[:space:]]+' '{print $1}')) && [[ ${#nvme_disks[@]} -eq 1 ]] && mkfs.ext4 -F ${nvme_disks[*]} && systemctl stop docker && mkdir -p /var/lib/kubelet/pods && mount ${nvme_disks[*]} /var/lib/kubelet/pods && chmod 750 /var/lib/docker && systemctl start docker
+      - nvme_disks=($(nvme list | grep "Amazon EC2 NVMe Instance Storage" | awk -F'[[:space:]][[:space:]]+' '{print $1}')) && [[ ${#nvme_disks[@]} -ge 2 ]] && mdadm --create --verbose /dev/md0 --level=0 --raid-devices=${#nvme_disks[@]} ${nvme_disks[*]} && mkfs.ext4 -F /dev/md0 && systemctl stop docker && mkdir -p /var/lib/kubelet/pods && mount /dev/md0 /var/lib/kubelet/pods && chmod 750 /var/lib/docker && systemctl start docker
+```
+
+
+**Benefits**
+
+* No need to mount the disk using Spark configurations or pod templates
+* Data generated by the application, will immediately be deleted at the pod termination. Data will be also purged in case of pod failures.
+* One time configuration for the node group
+
+**Cons**
+
+* Contention of disk resources if multiple jobs are allocated on the same node, as it’s not possible to properly re-partition resources between jobs
```

Comment:
If multiple jobs are allocated on the same EC2 instance, contention of disk resources will occur because it is not possible to allocate instance store volume resources across jobs